### PR TITLE
feat: add a spend notifier lambda

### PR DESCRIPTION
**Pull request by CalvinRodo**
# Summary | Résumé

Copy the lambda from the old organization and use it here
Move some data lookups to a global data lookups file

Related to: https://github.com/cds-snc/cds-aws-lz/issues/96

### CODE REVIEW

File: terragrunt/org_account/spend_notifier/lambda.tf [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  function_name = "spend_notifier"
+  role          = aws_iam_role.spend_notifier.arn
+  runtime       = "nodejs16.x"
+  handler       = "app.handler"
+  memory_size   = 512
+
+  filename         = data.archive_file.spend_notifier.output_path
+  source_code_hash = filebase64sha256(data.archive_file.spend_notifier.output_path)
+
+  reserved_concurrent_executions = 0
```
Comment (patheard) [inline terragrunt/org_account/spend_notifier/lambda.tf:20]: I'm assuming you don't want the spend notifier to start running yet?

File: terragrunt/org_account/spend_notifier/lambdas/spend_notifier/spend_notifier.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+                  "type": "plain_text",
+                  "text": `${account["Name"]} (${account["BU"]}): $${account["Cost"].toFixed(2).replace(/\d(?=(\d{3})+\.)/g, '$&,')} USD`,
+                },
+                "value": account["Name"]
+              } 
+            }),
+            "action_id": "overflow-action"
+          }
+        }
+        */
```
Comment (patheard) [inline terragrunt/org_account/spend_notifier/lambdas/spend_notifier/spend_notifier.js:76]: Can we remove this commented out block?

Comment (CalvinRodo) [reply terragrunt/org_account/spend_notifier/lambdas/spend_notifier/spend_notifier.js:76]: yeah not a problem that was left over from the code Max wrote didn't even review the lambda itself to be honest.

File: terragrunt/org_account/spend_notifier/lambda.tf (hunk 1 above)
Comment (CalvinRodo) [reply terragrunt/org_account/spend_notifier/lambda.tf:20]: I would like it start running asap

**Review by maxneuvians — APPROVED** [review]
Do it!

**Pull request merged**